specs/phase1/beacon-chain.md

```diff
@@ -12,150 +12,111 @@
+| `BLSCommitment` | `bytes48` | A G1 curve point |
 
 ## Configuration
 
-Configuration is not namespaced. Instead it is strictly an extension;
- no constants of phase 0 change, but new constants are adopted for changing behaviors.
-
 ### Misc
 
-| Name | Value |
-| - | - | 
-| `MAX_SHARDS` | `uint64(2**10)` (= 1024) |
-| `INITIAL_ACTIVE_SHARDS` | `uint64(2**6)` (= 64) |
-| `LIGHT_CLIENT_COMMITTEE_SIZE` | `uint64(2**7)` (= 128) |
-| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `uint64(2**3)` (= 8) | 
+| Name | Value | Notes |
+| - | - | - |
+| `MAX_SHARDS` | `uint64(2**10)` (= 1024) | Theoretical max shard count (used to determine data structure sizes) |
+| `INITIAL_ACTIVE_SHARDS` | `uint64(2**6)` (= 64) | Initial shard count |
+| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `uint64(2**3)` (= 2) | Gasprice may decrease/increase by at most exp(1 / this value) *per epoch* |
+| `MAX_SHARD_HEADERS_PER_SHARD` | `4` | |
+| `MAX_SHARD_HEADERS` | `MAX_SHARDS * MAX_SHARD_HEADERS_PER_SHARD` | |
 
 ### Shard block configs
 
-| Name | Value | Unit |
+| Name | Value | Notes |
 | - | - | - |
-| `MAX_SHARD_BLOCK_SIZE` | `uint64(2**20)` (= 1,048,576) | bytes |
-| `TARGET_SHARD_BLOCK_SIZE` | `uint64(2**18)` (= 262,144) |  bytes |
-| `SHARD_BLOCK_OFFSETS` | `List[uint64, 12]([1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233])` | - |
-| `MAX_SHARD_BLOCKS_PER_ATTESTATION` | `len(SHARD_BLOCK_OFFSETS)` | - |
-| `BYTES_PER_CUSTODY_CHUNK` | `uint64(2**12)` (= 4,096) | bytes |
-| `CUSTODY_RESPONSE_DEPTH` | `ceillog2(MAX_SHARD_BLOCK_SIZE // BYTES_PER_CUSTODY_CHUNK)` | - | 
+| `POINTS_PER_SAMPLE` | `uint64(2**3)` (= 8) | 31 * 8 = 248 bytes |
+| `MAX_SAMPLES_PER_BLOCK` | `uint64(2**11)` (= 2,048) | 248 * 2,048 = 507,904 bytes |
+| `TARGET_SAMPLES_PER_BLOCK` | `uint64(2**10)` (= 1,024) | 248 * 1,024 = 253,952 bytes |
 
-### Gwei values
+### Precomputed size verification points
 
 | Name | Value |
 | - | - |
-| `MAX_GASPRICE` | `Gwei(2**14)` (= 16,384) | Gwei | 
-| `MIN_GASPRICE` | `Gwei(2**3)` (= 8) | Gwei | 
+| `G2_ONE` | The G2 generator |
+| `SIZE_CHECK_POINTS` | Type `List[G2, MAX_SAMPLES_PER_BLOCK + 1]`; TO BE COMPUTED |
 
-### Initial values
+These points are the G2-side Kate commitments to `product[a in i...MAX_SAMPLES_PER_BLOCK] (X ** POINTS_PER_SAMPLE - w ** (revbit(a, MAX_SAMPLES_PER_BLOCK) * POINTS_PER_SAMPLE))` for each `i` in `[0...MAX_SAMPLES_PER_BLOCK]`, where `w` is the root of unity and `revbit` is the reverse-bit-order function. They are used to verify block size proofs. They can be computed with a one-time O(N^2/log(N)) calculation using fast-linear-combinations in G2.
 
-| Name | Value |
-| - | - |
-| `NO_SIGNATURE` | `BLSSignature(b'\x00' * 96)` | 
+### Gwei values
+
+| Name | Value | Unit | Description |
+| - | - | - | - |
+| `MAX_GASPRICE` | `Gwei(2**24)` (= 16,777,216) | Gwei | Max gasprice charged for an TARGET-sized shard block |  
+| `MIN_GASPRICE` | `Gwei(2**3)` (= 8) | Gwei | Min gasprice charged for an TARGET-sized shard block |
```

Comment:
Denominating in GWei seems coarse. Given that it's a 64 bit integer, it would be easy to increase resolution to Wei. Have you checked the dynamics of this when the gas price is at the lower end of the scale? Gas might become unresponsive/take longer to adjust

Reply:
It's not that coarse! Because it's not gwei per gas, it's gwei for the entire contents of the block. So the lowest possible gasprice is to literally pay (at current prices) $0.00000476 to fill an entire block.

Reply:
I'm just saying that at the lower limit it may behave erratically. Maybe we want a higher lower limit then?

Reply:
I just made a patch so that the gasprice delta is minimum 1, so it won't get stuck even at lower levels. Would that work? This is how the EIP 1559 proposal works on the eth1 side, and it seems a good idea to use similar formulas for both.

Reply:
I agree that this should pretty safely fix the issue at the low end of the spectrum. Question about the upper limit though -- that seems fairly low, only 0.017 ETH per block. That could very plausibly be breached. Current Eth1 blocks have already brought in several ETH per block.

(I guess this would mean that we have to check that a proposer has enough balance before creating a block)

Reply:
My current suggestion is to allow up to ~8 ETH gas (meaning ~16 ETH at full size) and block validators with insufficient balance from proposer selection.